### ISSUE TITLE
Remove engine.Status and replace with standard go error

### ISSUE THREAD

**Issue by crosbymichael** [docker/docker#11731]
**Problem**

In order to have type safe error checking we need to be able to propagate actual errors up to the caller.  Right now the engine has the concept of an `int` status code that is return.

**Solution**

Delete `engine.Status` and replace it's use with `error`.  This is a large refactoring that will touch many files but the end result is that we can handle errors in the standard go way and make type safe assertions.  

The main point where `engine.Status` is returned can be found: 

https://github.com/docker/docker/blob/master/engine/engine.go#L24

The status type can be found here and can be safely removed altogether:

https://github.com/docker/docker/blob/master/engine/job.go#L46

Code in consumers found in files like: 
https://github.com/docker/docker/blob/master/daemon/pause.go
can be changed to something like:

```go
+before
func (daemon *Daemon) ContainerPause(job *engine.Job) engine.Status {
	if len(job.Args) != 1 {
		return job.Errorf("Usage: %s CONTAINER", job.Name)
	}
	name := job.Args[0]
	container, err := daemon.Get(name)
	if err != nil {
		return job.Error(err)
	}
	if err := container.Pause(); err != nil {
		return job.Errorf("Cannot pause container %s: %s", name, err)
	}
	container.LogEvent("pause")
	return engine.StatusOK
}
+ after-------------->
func (daemon *Daemon) ContainerPause(job *engine.Job) error {
	if len(job.Args) != 1 {
		return fmt.Errorf("Usage: %s CONTAINER", job.Name)
	}
	name := job.Args[0]
	container, err := daemon.Get(name)
	if err != nil {
		return err
	}
	if err := container.Pause(); err != nil {
		return fmt.Errorf("Cannot pause container %s: %s", name, err)
	}
	container.LogEvent("pause")
	return nil
}
```

**Comment by tiborvass** [comment]
One of my biggest +1s :)

**Comment by jfrazelle** [comment]
+1 +1 +1

**Comment by vieux** [comment]
:dancer: :dancer: :dancer:

**Comment by runcom** [comment]
I'll gladly do it :smile:

**Comment by runcom** [comment]
#dibs

**Comment by icecrime** [comment]
<3

**Comment by tiborvass** [comment]
Closed by #11761
